### PR TITLE
Add integration tests for initContainers 

### PR DESCRIPTION
**Pull request by sankarpn**
Add initContainers under serverPod at spec, adminServer, Clusters, and managedServers level in domain.yaml and verify the initcontainers are created and runs to completion before the weblogic server pods are started.

### CODE REVIEW

**Comment by sankarpn** [comment]
The full Jenkins test suite passed.

http://wls-jenkins.us.oracle.com/view/weblogic-operator/job/weblogic-kubernetes-operator-javatest/2008/consoleFull

[INFO] Reactor Summary:
[INFO] 
[INFO] Build Tools 1.0 .................................... SUCCESS [  1.402 s]
[INFO] weblogic-kubernetes-operator 2.3.0 ................. SUCCESS [  2.960 s]
[INFO] json-schema 2.3.0 .................................. SUCCESS [ 17.036 s]
[INFO] jsonschema-maven-plugin Maven Mojo 2.3.0 ........... SUCCESS [  9.995 s]
[INFO] operator-model 2.3.0 ............................... SUCCESS [ 22.037 s]
[INFO] operator-swagger 2.3.0 ............................. SUCCESS [  5.530 s]
[INFO] operator-runtime 2.3.0 ............................. SUCCESS [ 53.804 s]
[INFO] operator-integration-tests 2.3.0 ................... SUCCESS [  06:44 h]
[INFO] installation-tests 2.3.0 ........................... SUCCESS [ 12.299 s]
[INFO] Project Reports 2.3.0 .............................. SUCCESS [  5.037 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  06:47 h
[INFO] Finished at: 2019-07-22T19:08:07-07:00
[INFO] ------------------------------------------------------------------------

**Comment by sankarpn** [comment]
Local runs passed too.

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Build Tools 1.0 .................................... SUCCESS [  3.062 s]
[INFO] weblogic-kubernetes-operator 2.3.0 ................. SUCCESS [ 10.274 s]
[INFO] json-schema 2.3.0 .................................. SUCCESS [ 18.243 s]
[INFO] jsonschema-maven-plugin Maven Mojo 2.3.0 ........... SUCCESS [ 10.626 s]
[INFO] operator-model 2.3.0 ............................... SUCCESS [ 22.132 s]
[INFO] operator-swagger 2.3.0 ............................. SUCCESS [  6.125 s]
[INFO] operator-runtime 2.3.0 ............................. SUCCESS [ 58.804 s]
[INFO] operator-integration-tests 2.3.0 ................... SUCCESS [  08:53 h]
[INFO] installation-tests 2.3.0 ........................... SUCCESS [  6.977 s]
[INFO] Project Reports 2.3.0 .............................. SUCCESS [  0.707 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  08:56 h
[INFO] Finished at: 2019-07-23T15:18:11-07:00
[INFO] ------------------------------------------------------------------------

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItInitContainers.java [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+
+    // Modify the original domain yaml to include restartVersion in admin server node
+    DomainCrd crd = new DomainCrd(originalYaml);
+    crd.addInitContNode("spec", null, null, "busybox", "foo");
+    crd.addInitContNode("adminServer", null, null, "busybox", "sleep");
+    crd.addInitContNode("clusters", "cluster-1", null, "busybox", "sleep");
+    String modYaml = crd.getYamlTree();
+    logger.info(modYaml);
+    testInitContainer(modYaml);
+    for (String pod : pods) {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItInitContainers.java]: you may need to check  TestUtils.checkCmdInLoop(cmd, "Init:0/1", pod[0]); to make sure only one initcontainer is started

File: integration-tests/USECASES.MD [hunk 2]
```diff
@@ -148,3 +148,14 @@ Configuration Overrides Usecases
 | Pod Templates | Use Case |
 | --- | --- |
 | Using pod templates/variables | Use DOMAIN_UID, DOMAIN_NAME, DOMAIN_HOME, SERVER_NAME, LOG_HOME variables at domain level and CLUSTER_NAME at cluster level. Bring up the domain and make sure domain started successfully when pod templates are used |
+
+| Init Container | Use Case |
+| --- | --- |
+| Add initContainers to domain | Add a initContainers object to spec level and verify the init containers are created for weblogic server pods prior to starting it and runs to completion and then weblogic pod are started |
```
Comment (rjeberhard) [inline integration-tests/USECASES.MD]: weblogic -> WebLogic

Comment (sankarpn) [reply integration-tests/USECASES.MD]: Done

**Pull request merged**